geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/records/package.scala

```diff
@@ -323,7 +341,10 @@ package object records {
     private abstract class DynamicEvaluation[T](prop: PropertyDescriptor) extends PropertyExtractor[T] {
       override def apply(context: PropertyContext, variables: java.util.Map[String, String]): T = {
         val p = context.getProperty(prop)
-        wrap(if (p.isSet) { p.evaluateAttributeExpressions(variables).getValue } else { null })
+        val value = if (!p.isSet) { null } else {
+          Option(p.evaluateAttributeExpressions(variables).getValue).filter(_.nonEmpty).orNull
```

Comment:
Is the change in logic here that an empty value is mapped to null?

Reply:
yeah, since there's a default nifi expression in the field now, if the variable isn't set it will end up an empty string instead of null, but we don't want to actually use it.